apps/condo/domains/acquiring/utils/billingCentrifuge.js

```diff
@@ -0,0 +1,343 @@
+        orderGroup.push(d)
+        set(res, order, orderGroup)
+        groupKeys.add(order)
+
+        totalDistributionAmount = totalDistributionAmount.plus(d.amount)
+        return res
+    }, {})
+
+    const sortedGroupKeys = uniq(Array.from(groupKeys.values())).sort()
+
+    let totalAppliedAmount = Big(0) // including fees
+
+    const appliedAmounts = {}
+    const appliedFeeAmounts = {}
+    for (const split of appliedSplits) {
+        if (!split.recipient) continue
+
+        totalAppliedAmount = totalAppliedAmount.plus(split.amount || 0).plus(split.feeAmount || 0)
+
+        const key = createRecipientKey(split.recipient)
+
+        const appliedAmount = get(appliedAmounts, key, Big(0))
+        appliedAmounts[key] = appliedAmount.plus(split.amount || 0)
+
+        const appliedFeeAmount = get(appliedFeeAmounts, key, Big(0))
+        appliedFeeAmounts[key] = appliedFeeAmount.plus(split.feeAmount || 0)
+    }
+
+    let restUndistributedAmount = Big(paymentAmount)
+
+    /** @type TSplit[] */
+    const splits = []
+
+    for (const order of sortedGroupKeys) {
+        const g = groupedDistributions[order]
+
+        // Each group must contain a SINGLE item with vor=true
+        if (!hasSingleVorItem(g)) {
+            throw new Error(`Group ${order} does not contains a SINGLE element with vor=true`)
+        }
+
+        const noopForGroupAmount = g.reduce((res, d) => {
+            const key = createRecipientKey(d.recipient)
+            return res
+                .plus(get(appliedAmounts, key, Big(0)))
+                .plus(get(appliedFeeAmounts, key, Big(0)))
+        }, Big(0))
+        let needToSplitToGroupAmount = g.reduce((res, d) => res.plus(Big(d.amount)), Big(0)).minus(noopForGroupAmount)
+
+        if (needToSplitToGroupAmount.lte(0)) {
+            needToSplitToGroupAmount = Big(0)
+            continue
+        }
+        const hasEnoughAmountForGroup = needToSplitToGroupAmount.lte(restUndistributedAmount)
+        const undistributedAmountAvailableForGroup = Big(restUndistributedAmount)
+
+        /**
+         * @type {TDistribution[]}
+         * The victim of the rounding operation MUST be the last item within the group
+         */
+        const sortedGroup = g.sort((a, b) => a.vor ? 1 : -1)
+
+        for (let i = 0; i < sortedGroup.length; i++) {
+            const d = sortedGroup[i]
+            const recipientKey = createRecipientKey(d.recipient)
+            // The rest needed amount for particular recipient
+            let distributionAmount = Big(d.amount)
+                .minus(get(appliedAmounts, recipientKey, Big(0)))
+                .minus(get(appliedFeeAmounts, recipientKey, Big(0)))
+
+            let share = hasEnoughAmountForGroup ? distributionAmount : distributionAmount.div(needToSplitToGroupAmount).times(undistributedAmountAvailableForGroup)
+
+            if (share.lte(0)) {
+                continue
+            }
+
+            const roundedShare = share.round(decimalPlaces, 1)
```

Comment:
<img width="736" alt="image" src="https://github.com/user-attachments/assets/f922b3d2-01a8-4bd4-9686-a14f4363c8da" />

Reply:
Changed to `Big.roundHalfUp`